### DISCUSSION TITLE
`.conf` syntax highlighting

### DISCUSSION THREAD

**Discussion by anoduck** [helix-editor/helix#10776] [Q&A] [answered]
This has to have been tackled before, but the number one most irritating thing about Helix is finding a language syntax that appropriately fits configuration files or "ini" files. I am not talking about the less common variants where `;`, `:`, or `'` are used to denote comments. I am referring to the most common form, where `#` represents a commented line. Neither the ssh-client-config syntax nor the git-config syntax work in this respect.

**Comment by yanniskatsaros** [comment]
@anoduck As far as I can tell, Helix has a pre-configured INI section in its default `languages.toml` (see the [source file](https://github.com/helix-editor/helix/blob/master/languages.toml#L2679-L2713), reproduced below as well).

I haven't tried this personally, but you could add the following to your `~/.config/helix/languages.toml` (or wherever you configure Helix) **see the examples provided in the `file-types` array**

```toml
[[language]]
name = "ini"
scope = "source.ini"
file-types = [
  "ini",
  # Systemd unit files
  "service",
  "automount",
  "device",
  "mount",
  "path",
  "service",
  "slice",
  "socket",
  "swap",
  "target",
  "timer",
  # Podman quadlets
  "container",
  "volume",
  "kube",
  "network",
  { glob = ".editorconfig" },
  { glob = "rclone.conf" },
  "properties",
  "cfg",
  "directory",

  # === your specific .conf files or all ".conf" files ===
  # { glob = "some_file.conf" },
  # "conf",
]
injection-regex = "ini"
comment-token = "#"
indent = { tab-width = 4, unit = "\t" }

[[grammar]]
name = "ini"
source = { git = "https://github.com/justinmk/tree-sitter-ini", rev = "87176e524f0a98f5be75fa44f4f0ff5c6eac069c" }
```

(note: I also updated the Tree Sitter grammar to the latest `git` hash)

**Reply by anoduck** [reply]
@yanniskatsaros Helix rocks. I find myself using it more and more all the time. 

What about nginx.conf and apache2.conf? These are both `conf` files edited often.

**Reply by yanniskatsaros** [reply, marked as answer]
@anoduck a quick search for "tree sitter nginx conf" on Google turned up this repo: https://gitlab.com/joncoole/tree-sitter-nginx

It is hosted on GitLab and I see most examples for `[[grammar]]` use GitHub links, but you can give it a shot. I'm not sure if this would work / haven't looked into the docs, but you could give something like this a shot:

```toml
[[grammar]]
name = "nginx.conf"
source = { git = "https://gitlab.com/joncoole/tree-sitter-nginx", rev = "b4b61db443602b69410ab469c122c01b1e685aa0" }
```

which should provide syntax highlighting for NGINX conf files.

Again, no idea if this would work since most grammar sections list a file extension, not a specific file name so you'd have to just give it a shot to see if that would work.

**Reply by anoduck** [reply]
@yanniskatsaros You are a lifesaver. God bless you. This will make living with that remote nginx server so much better. Cheers.
